libs/base/ui/structure/heading/src/heading.util.ts

```diff
@@ -0,0 +1,72 @@
+import { Size } from '@spryker-oryx/utilities';
+import { TemplateResult } from 'lit';
+import { HeadingDirectiveOptions, HeadingTypography } from './heading.model';
+
+/**
+ * Utility function to render a heading.
+ */
+export const renderHeading = (
+  content?: string | TemplateResult,
+  options?: HeadingDirectiveOptions,
+  attributes?: string
+): string => {
+  const { tag } = options ?? {};
+  const style = getHeaderStyles(options, extractStyleAttributes(attributes));
+  let tagName: string;
+
+  switch (tag) {
+    case (HeadingTypography.Caption,
+    HeadingTypography.Small,
+    HeadingTypography.Subtitle,
+    HeadingTypography.SubtitleSmall):
+      tagName = 'span';
+      break;
+    case HeadingTypography.Bold:
+      tagName = 'b';
+      break;
+    default:
+      tagName = tag ?? 'span';
+  }
+
+  return `<${tagName} style="${style}" ${
```

Comment:
we are building string here, for the whole TAG, and passing it into `unsafeStatic` level above, which makes it less efficient and more memory efficient, especially if header will change often at runtime.

https://lit.dev/docs/templates/expressions/#static-expressions
> Changing the value of static expressions is expensive. Expressions using literal values should not change frequently, as they cause a new template to be re-parsed and each variation is held in memory.

maybe we should consider use unsafeStatic only for the parts we really need, and return template instead? Then change of content will not cause full heading structure regeneration?
(but then directive is less usefull for rich text component)

Reply:
Refactored.